assets/js/components/notifications/dashboard-auth-alert.js

```diff
@@ -28,18 +28,26 @@ import Notification from '../notifications/notification';
 
 const DashboardAuthAlert = () => {
 	const { admin: { connectURL } } = global.googlesitekit;
-	const { currentAdminPage } = global.googlesitekit.admin;
-	const product = currentAdminPage
-		.replace( /googlesitekit|module|-/g, ' ' )
-		.replace( /(^\w{1})|(\s{1}\w{1})/g, ( match ) => match.toUpperCase() )
-		.trim();
-
+	const { unsatisfiedScopes } = global.googlesitekit.setup;
+	let message = '';
+	const missingScopes = [ ...new Set( unsatisfiedScopes.map( ( item ) => item.match( /https:\/\/www.googleapis.com\/auth\/([a-z]+)/ )[ 1 ] ) ) ];
+	const moduleNames = missingScopes.map( ( module ) => module.charAt( 0 ).toUpperCase() + module.slice( 1 ) );
+	if ( 1 < moduleNames.length ) {
+		/* translators: %1$s: Product name */
+		message = sprintf( __( 'Site Kit can’t access the some relevant data because you haven’t granted all API scopes requested during setup. To use Site Kit, you’ll need to redo the setup for: %1$s – make sure to approve all API scopes at the authentication stage.', 'google-site-kit' ), moduleNames.join( ', ' ) );
```

Comment:
* In this case, let's use a single `%s` placeholder as only a single value is used once.
* The translators comment is slightly inaccurate as it is a list of multiple product names
* The list separator should be translatable as well. See https://github.com/google/site-kit-wp/blob/f4967c34d3d91eefddb9446b73614e1341e48b2f/includes/Core/Util/Debug_Data.php#L271-L272
* Let's split the function call over a few lines to keep it from running overly long and "hiding" the parameters filling the placeholders. (same applies to other `sprintf` calls here)